ui/src/app/applications/components/utils.tsx

```diff
@@ -945,6 +945,53 @@ export function getPodStateReason(pod: appModels.State): {message: string; reaso
     return {reason, message};
 }
 
+export const getPodReadinessGatesState = (pod: appModels.State): {nonExistingConditions: string[]; failedConditions: string[]} => {
+    if (!pod.spec?.readinessGates?.length) {
+        return {
+            nonExistingConditions: [],
+            failedConditions: []
+        };
+    }
+
+    const existingConditions = new Map<string, boolean>();
+    const podConditions = new Map<string, boolean>();
+
+    const podStatusConditions = pod.status?.conditions || [];
+
+    for (const condition of podStatusConditions) {
+        existingConditions.set(condition.type, true);
+        // priority order of conditions
+        // eg. if there are multiple conditions set with same name then the one which comes first is evaluated
+        if (podConditions.has(condition.type)) {
+            continue;
+        }
+
+        if (condition.status === 'False') {
+            podConditions.set(condition.type, false);
+        } else if (condition.status === 'True') {
+            podConditions.set(condition.type, true);
+        }
+    }
+
+    const nonExistingConditions: string[] = [];
+    const failedConditions: string[] = [];
+
+    const readinessGates: appModels.ReadinessGate[] = pod.spec?.readinessGates || [];
+
+    for (const readinessGate of readinessGates) {
+        if (!existingConditions.has(readinessGate.conditionType)) {
+            nonExistingConditions.push(readinessGate.conditionType);
+        } else if (podConditions.get(readinessGate.conditionType) === false) {
+            failedConditions.push(readinessGate.conditionType);
+        }
+    }
```

Comment:
Leaving this here for my own future reference:

Looks like a readiness gate may have either a corresponding “success” condition (meaning things are fine), a “failed” condition (meaning things are not fine), or no condition at all (meaning things are not yet fine).